.github/workflows/update_dataset_repos.yml

```diff
@@ -0,0 +1,29 @@
+name: Update dataset repos
+on:
+  push:
+    branches:
+      - 'main'
+jobs:
+  update-dataset-repo:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fork:
+          - neurodatascience/mr_proc-ppmi
+          - neurodatascience/mr_proc-qpn
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ matrix.fork }}
+      - name: Get upstream changes
+        run: |
+          git remote add upstream https://github.com/neurodatascience/mr_proc.git
+          git pull upstream main
+          git fetch upstream main:upstream-main
+          git reset --hard upstream-main
```

Comment:
while a git reset --hard is always attractive, not sure I see if it is actually  needed just after the fetch upstream main:upstream-main : will that branch pre-exists and may contain some stagged or non committed files ? or some commits ahead of upstream main ?  or is it to make sure everything is clean there ?

Reply:
I think it's because `neurodatascience/mr_proc-ppmi` and `neurodatascience/mr_proc-qpn` will have additional dataset  specific files (consequently commits) which are not in `neurodatascience/mr_proc`.
@michellewang correct me if that's not the reason!

Reply:
@jbpoline @nikhil153 I could be wrong (I haven't used GitHub Actions before and am still not too familiar), but I think the `actions/checkout@v3` action creates a fresh clone of the repository, so there shouldn't be any staged or non-committed files. There might be commits ahead of upstream main though because the forks (`neurodatascience/mr_proc-ppmi` etc.) will have additional dataset-specific files, like @nikhil153 said. In that case the `reset --hard` still makes sense (I think) because we only want to get the changes that were added to `neurodatascience/mr_proc`, not the PPMI/QPN forks. Then it will create/update a PR to add these changes to `neurodatascience/mr_proc-ppmi`/`neurodatascience/mr_proc-qpn`, and we will have to manually resolve merge conflicts if there are any, then merge to the main branch of the fork.

I've tried removing the `git reset --hard upstream-main` line and it didn't work, the PRs got closed automatically without merging (not 100% sure why). There might still be other ways to achieve the same results without `git reset --hard`.

Reply:
I see (I think :) ) - the workflow makes sense to me !